backend/api/serializers/ExclusionReport.py

```diff
@@ -0,0 +1,165 @@
+"""
+    REST API Documentation for the NRS TFRS Credit Trading Application
+
+    The Transportation Fuels Reporting System is being designed to streamline
+    compliance reporting for transportation fuel suppliers in accordance with
+    the Renewable & Low Carbon Fuel Requirements Regulation.
+
+    OpenAPI spec version: v1
+
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+from rest_framework import serializers
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.relations import SlugRelatedField
+
+from api.models.ApprovedFuel import ApprovedFuel
+from api.models.CompliancePeriod import CompliancePeriod
+from api.models.ComplianceReport import \
+    ComplianceReport, ComplianceReportStatus
+from api.models.ExclusionReportAgreement import \
+    ExclusionAgreement, ExclusionAgreementRecord
+from api.models.TransactionType import TransactionType
+from api.serializers.ComplianceReport import \
+    ComplianceReportStatusSerializer, ComplianceReportTypeSerializer
+from api.serializers import \
+    OrganizationMinSerializer, CompliancePeriodSerializer
+
+
+class ExclusionAgreementRecordSerializer(serializers.ModelSerializer):
+    transaction_type = SlugRelatedField(
+        slug_field='the_type', queryset=TransactionType.objects.all())
+    fuel_type = SlugRelatedField(
+        slug_field='name', queryset=ApprovedFuel.objects.all())
+
+    class Meta:
+        model = ExclusionAgreementRecord
+        fields = (
+            'transaction_partner', 'postal_address', 'quantity', 'quantity',
+            'quantity_not_sold', 'transaction_type', 'fuel_type'
+        )
+
+
+class ExclusionAgreementSerializer(serializers.ModelSerializer):
+    records = ExclusionAgreementRecordSerializer(many=True, required=False)
+
+    class Meta:
+        model = ExclusionAgreement
+        fields = ('records',)
+
+
+class ExclusionReportDetailSerializer(serializers.ModelSerializer):
+    """
+    Detail Serializer for the Exclusion Report
+    """
+    status = ComplianceReportStatusSerializer(read_only=True)
+    type = ComplianceReportTypeSerializer(read_only=True)
```

Comment:
My IDE thinks type is a reserved word. It seems to work but maybe rename it for clarity.

Reply:
ill take a look at this. i prefer to keep the name similar to the one in the model to lessen confusion

Reply:
Ya is there a way for the IDE to ignore this instead? type is quite spread around our Compliance Report serializers and we're into this too deep to make changes.